### PR TITLE
[`Refactor`] Appended `Async` to async methods names

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Change Log
- Appended `Async` to async methods names
- Updated `.editorconfig` with new rules, nothing in code changed besides above.

### Needs #3214 to work

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shargon — CHANGES_REQUESTED** [review]
We use reflection for smart contract methods names, this can't be done automatically

**Comment by cschuchardt88** [comment]
Yes i know, I added _see below_ to fix that issue.

https://github.com/neo-project/neo/blob/9800e0d7170f457412b6faf44cde51c65f2b711d/src/Neo/SmartContract/Native/ContractManagement.cs#L13

File: .editorconfig [hunk 1]
```diff
@@ -48,15 +48,30 @@ indent_size = 2
...
 
+# .github workflows
+[*.{yaml, yml}]
+end_of_line = lf
+indent_size = 2
+
 # Dotnet code style settings:
 [*.{cs,vb}]
+
+// Missing XML comment for publicly visible type or member
```
Comment (AnnaShaleva) [inline .editorconfig:59]: Reformat the comment to follow the overall style? It starts with `//`.

File: .editorconfig [hunk 2]
```diff
@@ -48,15 +48,30 @@ indent_size = 2
...
+
 # Member can be made 'readonly'
 csharp_style_prefer_readonly_struct_member = true
 dotnet_diagnostic.IDE0251.severity = warning
 dotnet_diagnostic.IDE0044.severity = warning
 
 dotnet_diagnostic.CS1591.severity = silent
 
+# Avoid problematic synchronous waits
+dotnet_diagnostic.VSTHRD002.severity = none
```
Comment (AnnaShaleva) [inline .editorconfig:70]: Change the level to warning?

File: src/Neo.GUI/Neo.GUI.csproj [hunk 3]
```diff
@@ -19,7 +20,7 @@
   <ItemGroup>
     <Content Include="neo.ico" />
   </ItemGroup>
-
+
```
Comment (AnnaShaleva) [inline src/Neo.GUI/Neo.GUI.csproj:23]: Useless change.

File: src/Neo/SmartContract/Native/ContractManagement.cs [hunk 4]
```diff
@@ -187,13 +188,16 @@ public IEnumerable<ContractState> ListContracts(DataCache snapshot)
         }
 
         [ContractMethod(RequiredCallFlags = CallFlags.All)]
+
```
Comment (AnnaShaleva) [inline src/Neo/SmartContract/Native/ContractManagement.cs:191]: Useless change.

File: src/Neo/SmartContract/Native/ContractManagement.cs [hunk 5]
```diff
@@ -187,13 +188,16 @@ public IEnumerable<ContractState> ListContracts(DataCache snapshot)
...
         private ContractTask<ContractState> Deploy(ApplicationEngine engine, byte[] nefFile, byte[] manifest)
         {
             return Deploy(engine, nefFile, manifest, StackItem.Null);
         }
 
         [ContractMethod(RequiredCallFlags = CallFlags.All)]
+#pragma warning disable IDE1006 // Naming Styles
         private async ContractTask<ContractState> Deploy(ApplicationEngine engine, byte[] nefFile, byte[] manifest, StackItem data)
+#pragma warning restore IDE1006 // Naming Styles
```
Comment (AnnaShaleva) [inline src/Neo/SmartContract/Native/ContractManagement.cs:200]: Deserves more detailed comment about `Async` suffix.

File: src/Neo/SmartContract/Native/FungibleToken.cs [hunk 6]
```diff
@@ -70,7 +72,9 @@ protected override void OnManifestCompose(ContractManifest manifest)
             manifest.SupportedStandards = new[] { "NEP-17" };
         }
 
+#pragma warning disable IDE1006 // Naming Styles
         internal async ContractTask Mint(ApplicationEngine engine, UInt160 account, BigInteger amount, bool callOnPayment)
+#pragma warning restore IDE1006 // Naming Styles
```
Comment (AnnaShaleva) [inline src/Neo/SmartContract/Native/FungibleToken.cs:77]: Why? This method is not included into manifest.

File: src/Neo/SmartContract/Native/FungibleToken.cs [hunk 7]
```diff
@@ -83,7 +87,9 @@ internal async ContractTask Mint(ApplicationEngine engine, UInt160 account, BigI
             await PostTransferAsync(engine, null, account, amount, StackItem.Null, callOnPayment);
         }
 
+#pragma warning disable IDE1006 // Naming Styles
         internal async ContractTask Burn(ApplicationEngine engine, UInt160 account, BigInteger amount)
+#pragma warning restore IDE1006 // Naming Styles
```
Comment (AnnaShaleva) [inline src/Neo/SmartContract/Native/FungibleToken.cs:92]: As far as this one.

File: .editorconfig (hunk 2 above)
Comment (cschuchardt88) [reply .editorconfig:70]: This can break code; if you do `dotnet format`.

**Comment by cschuchardt88** [comment]
Why is this taking too long?

File: src/Neo.GUI/GUI/UpdateDialog.cs [hunk 8]
```diff
@@ -47,7 +50,7 @@ private void linkLabel2_LinkClicked(object sender, LinkLabelLinkClickedEventArgs
             Process.Start(download_url);
         }
 
-        private async void button2_Click(object sender, EventArgs e)
+        private async void Button2_Click(object sender, EventArgs e)
```
Comment (shargon) [inline src/Neo.GUI/GUI/UpdateDialog.cs:53]: This can be end with Async

File: src/Neo/SmartContract/Native/ContractManagement.cs [hunk 9]
```diff
@@ -194,6 +196,7 @@ private ContractTask<ContractState> Deploy(ApplicationEngine engine, byte[] nefF
 
         [ContractMethod(RequiredCallFlags = CallFlags.All)]
         private async ContractTask<ContractState> Deploy(ApplicationEngine engine, byte[] nefFile, byte[] manifest, StackItem data)
+
```
Comment (shargon) [inline src/Neo/SmartContract/Native/ContractManagement.cs:199]:
```suggestion
```

File: src/Neo/SmartContract/ApplicationEngine.Contract.cs [hunk 10]
```diff
@@ -147,7 +148,7 @@ internal protected UInt160 CreateMultisigAccount(int m, ECPoint[] pubKeys)
         /// The implementation of System.Contract.NativeOnPersist.
         /// Calls to the <see cref="NativeContract.OnPersistAsync"/> of all native contracts.
         /// </summary>
-        protected internal async void NativeOnPersistAsync()
+        protected internal async Task NativeOnPersistAsync()
```
Comment (shargon) [inline src/Neo/SmartContract/ApplicationEngine.Contract.cs:151]: I'm not sure if need to use `ContractTask` in all of the renamed method related to native contracts

Comment (cschuchardt88) [reply src/Neo/SmartContract/ApplicationEngine.Contract.cs:151]: Using `async void` could have the process lock up.

File: src/Neo.GUI/GUI/UpdateDialog.cs (hunk 8 above)
Comment (cschuchardt88) [reply src/Neo.GUI/GUI/UpdateDialog.cs:53]: I'm not to worried about `neo-gui`.

File: src/Neo/SmartContract/ApplicationEngine.Contract.cs (hunk 10 above)
Comment (shargon) [reply src/Neo/SmartContract/ApplicationEngine.Contract.cs:151]: I mean `ContractTask` instead of `Task` we have some `is ContractTask`

Comment (cschuchardt88) [reply src/Neo/SmartContract/ApplicationEngine.Contract.cs:151]: @shargon you keep breaking my PRs.... lol 🥇 

What i was saying is it was `void` so `no` you don't need to change to `ContractTask`.

**Review by Jim8y — APPROVED** [review]

File: src/Neo/SmartContract/ApplicationEngine.Contract.cs (hunk 10 above)
Comment (cschuchardt88) [reply src/Neo/SmartContract/ApplicationEngine.Contract.cs:151]: Plus I tried `ContractTask` and it didn't work.

**Comment by vncoelho** [comment]
> We use reflection for smart contract methods names, this can't be done automatically

Was this comment from @shargon solved of replied?

File: src/Neo.GUI/GUI/UpdateDialog.cs (hunk 8 above)
Comment (shargon) [reply src/Neo.GUI/GUI/UpdateDialog.cs]: It's a mono-repo

**Comment by cschuchardt88** [comment]
> > We use reflection for smart contract methods names, this can't be done automatically
> 
> Was this comment from @shargon solved or replied?

I say resolved, because it's been tested with the `unit` `tests`. Everything working fine.

File: src/Neo/SmartContract/Native/NativeContract.cs [hunk 11]
```diff
@@ -361,7 +362,7 @@ public static NativeContract GetContract(UInt160 hash)
             return contract;
         }
 
-        internal async void Invoke(ApplicationEngine engine, byte version)
+        internal async Task InvokeAsync(ApplicationEngine engine, byte version)
```
Comment (shargon) [inline src/Neo/SmartContract/Native/NativeContract.cs:365]: I'm worry in line 392, when we check that it must be a `ContractTask` and now some method returns `Task`

Comment (cschuchardt88) [reply src/Neo/SmartContract/Native/NativeContract.cs:365]: I already tested this. Plus if it was void then it needs to be task. I also tried `ContractTask` and didn't work.

**Pull request closed without merging**